client/src/pages/WaitingRoom.js

```diff
@@ -89,13 +91,62 @@ function WaitingRoom(props) {
   const [openRoom, setOpenRoom] = useState(true);
   const [fullUrlPath, setFullUrlPath] = useState("");
   const history = useHistory();
-
+  const { user } = useContext(UserContext);
+  const [participants, setParticipants] = useState([]);
+  const [inRoom, setInRoom] = useState(false);
+  const [startEnabled, setstartEnabled] = useState(false);
   useEffect(() => {
+    handleInRoom();
+    if (inRoom) {
+      console.log("joiningRoom");
+      socket.emit(
+        "joinInterviewLobby",
+        {
+          id: props.match.params.id,
+          name: user.firstName + " " + user.lastName,
+          userId: user.id,
+        },
+        function (confimation) {
+          console.log(confimation);
+          if (!confimation) history.push("/dashboard");
+        }
+      );
+    }
     setFullUrlPath(window.location.href);
-  }, []);
+    return () => {
+      if (inRoom) {
+        console.log("leaving room");
+        socket.emit("leaveRoom", {
+          id: props.match.params.id,
+          name: user.firstName + " " + user.lastName,
+          userId: user.id,
+        });
+      }
+    };
+  }, [user]);
 
   function handleStartButtonClick() {}
 
+  useEffect(() => {
+    socket.on("joinedRoom", (data) => {
+      let part = [];
+      part.firstName = data;
```

Comment:
Is this necessary?

Reply:
for now yes but when the sockets sends more info then will remove